### ISSUE TITLE
Normalization

### ISSUE THREAD

**Issue by vibaraz** [hm-choi/uni-henn#36]
Hello, I'm currently working with the uni-henn encryption library and was wondering if it's possible to integrate batch normalization—or any other type of normalization—into the neural network. Thank you for your help.

**Comment by hm-choi** [comment]
Thank you for your attention.

Batch Normalization in inference only requires constant multiplication and addition, so you can attach the normalization layer using SEALPython API’s. 

![Image](https://github.com/user-attachments/assets/0670400a-93cb-4c50-9443-068a00dd8dd4)

**Comment by vibaraz** [comment]
Thank you for the explanation, hm-choi. I have successfully implemented batch normalization and now, I'm exploring whether it's possible to incorporate residual connections using Uni-HENN. Could you please share any insights or examples on how to effectively implement them? Thank you for your help.

**Comment by hm-choi** [comment]
> Thank you for the explanation, hm-choi. I have successfully implemented batch normalization and now, I'm exploring whether it's possible to incorporate residual connections using Uni-HENN. Could you please share any insights or examples on how to effectively implement them? Thank you for your help.

Thank you for your attention.
Sorry for the late reply.
Our architecture does not consider the residual connections, but you can refer to the paper https://eprint.iacr.org/2024/460.pdf to construct a residual connection.

The MS SEAL does not support bootstrapping, so you couldn't implement deep layer, so we recommend using another library such as lattigo or openfhe that supports bootstrapping. 

Thanks.
